source/02_algo/multi_agent_cooperation_rl_eng.rst

```diff
@@ -0,0 +1,100 @@
+   :scale: 50 %
+
+
+In MARL, the policy controls multiple agents to interact with the environment simultaneously, and the goal is still to maximize the cumulative reward that can be obtained. At this time, the transition of the global state of the environment and the reward value are related to the joint action of all agents rather than the action of a single agent. Therefore, in the process of policy learning, the update of each agent's policy needs to consider the current policies of other agents.
+
+.. image:: images/MARL_summary.png
+   :align: center
+   :scale: 50 %
+
+In this figure, system represents a multi-agent environment:math:`Agent_i` represents the ith agent,:math:`a_i` represents the action taken by the ith agent,:math:`r_i` represents the local reward obtained by the ith agent.
+During the training process, each agent interacts with the environment separately, and the system feeds back a joint reward.
+
+
+In general, the main difference between multi-agent reinforcement learning and single-agent reinforcement learning lies in the following four points:
+
+  1. The non-stationary of the environment: While the agent is making a decision, other agents are also taking actions, and the change of the environment state is related to the joint action of all the agents. Therefore, the value evaluation of the actions of a single agent will change with the actions of other agents, which will lead to non-stationary in MARL training.
+
+  2. Limitations of information acquisition by agents: In some environments (such as SMAC), each agent may not be able to obtain global state information, but only local observation information. However, it is impossible to know the observation information, actions and other information of other agents.
+
+  3. Individual goal consistency: The goal of each agent may be the optimal global return or the optimal local return.
+
+  4. Scalability: Large-scale multi-agent systems may involve high-dimensional state space and action space, which poses certain challenges for model expression ability, algorithm learning ability and hardware computing power in real scenarios.
+
+
+
+Research Direction
+-----------------------
+For the MARL cooperation task, the simplest idea is to directly apply the single-agent reinforcement learning method to the multi-agent system, that is, each agent regards other agents as factors in the environment, and still learns according to the single agent way to update the policy by interacting with the environment; This is the idea of independent Q-learning, independent PPO methods, but due to the non-stationarity of the environment and the locality of agent observations, these methods are difficult to achieve good results.
+
+At present, MARL cooperation mainly adopts the method of CTDE (centralized training and decentralized execute), and there are two main types of solutions, Valued-based MARL and Actor-Critic MARL. For details, please refer to the following figure:
```

Comment:
At present, the cooperative tasks in MARL mainly adopt the CTDE paradigm (centralized training and decentralized execution), which mainly include two types of learning methods, Valued-based MARL and Actor-Critic MARL. Please refer to Figure 2 for details: